hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheProxy.java

```diff
@@ -297,6 +339,32 @@ private void putToAllPartitionsAndWaitForCompletion(List<Map.Entry<Data, Data>>[
         }
     }
 
+    private void setTTLAllPartitionsAndWaitForCompletion(List<Data>[] keysPerPartition, Data expiryPolicy) {
+        List<Future> futures = new ArrayList<Future>(keysPerPartition.length);
+        for (int partitionId = 0; partitionId < keysPerPartition.length; partitionId++) {
+            List<Data> keys = keysPerPartition[partitionId];
+            if (keys != null) {
+                Operation operation = operationProvider.createSetExpiryPolicyOperation(keys, expiryPolicy);
+                futures.add(invoke(operation, partitionId, true));
+            }
+        }
+
+        Throwable error = null;
+        for (Future future: futures) {
+            try {
+                future.get();
+            } catch (Throwable t) {
+                logger.finest("Error occured while batch operation!", t);
+                if (error == null) {
+                    error = t;
+                }
+            }
+        }
+        if (error != null) {
+            throw rethrow(error);
+        }
```

Comment:
To not to write the part between lines 352-365, maybe you can enhance `FutureUtil#waitForever`.